services/repository/transfer.go

```diff
@@ -337,7 +337,10 @@ func changeRepositoryName(ctx context.Context, repo *repo_model.Repository, newR
 	}
 
 	if err = gitrepo.RenameRepository(ctx, repo,
-		repo_model.StorageRepo(repo_model.RelativePath(repo.OwnerName, newRepoName))); err != nil {
+		repo_model.StorageRepo{
+			RelativePath:     repo_model.RelativePath(repo.OwnerName, newRepoName),
+			ObjectFormatName: repo.ObjectFormatName,
```

Comment:
You see, I have told you, it is not right.

I do not see why your design would work without buggy behaviors.

* https://github.com/go-gitea/gitea/pull/33911#discussion_r2018004373

Reply:
0d941b5 reverted the new interface.